bibtex-completion.el

```diff
@@ -478,6 +493,30 @@ for string replacement."
 (defvar bibtex-completion-cached-notes-keys nil
   "A cache storing notes keys obtained when the bibliography was last parsed.")
 
+(defun bibtex-completion--read ()
+  "Select BibTeX entries in completion system."
+  ;; define a completion function that defaults to completing-read, but can be overridden
+  (list (cdr (assoc (bibtex-completion--completing-read) (bibtex-completion--get-candidates)))))
```

Comment:
Seems like the `bibtex-completion--get-candidates` call should be moved down into the `bibtex-completion--completing-read` function, given that it's only used there (it's not need in the ivy or helm packages).